extensions/ql-vscode/src/vscode-tests/cli-integration/queries.test.ts

```diff
@@ -100,4 +100,27 @@ describe('Queries', function() {
     }
   });
 
+  // Asserts a fix for bug https://github.com/github/vscode-codeql/issues/733
+  it.only('should restart the database and run a query', async () => {
+    try {
+      await commands.executeCommand('codeQL.restartQueryServer');
+      const queryPath = path.join(__dirname, 'data', 'simple-query.ql');
+      const result = await compileAndRunQueryAgainstDatabase(
+        cli,
+        qs,
+        dbItem,
+        false,
+        Uri.file(queryPath),
+        progress,
+        token
+      );
+
+      // this message would indicate that the databases were not properly reregistered
+      expect(result.result.message).not.to.eq('No result from server');
```

Comment:
Can you also check that the `result` status is successful? Otherwise we could change the message and break the behaviour without failing this test.

Reply:
Sure...I can do something here.